src/SignalR/clients/java/signalr/core/src/main/java/com/microsoft/signalr/HubConnection.java

```diff
@@ -425,48 +311,24 @@ public Completable start() {
             }).subscribe(() -> {
                 localStart.onComplete();
             }, error -> {
-                hubConnectionStateLock.lock();
-                hubConnectionState = HubConnectionState.DISCONNECTED;
-                hubConnectionStateLock.unlock();
+                this.state.lock.lock();
```

Comment:
The lock inside of changeState makes this redundant I think. If changeState somehow threw because of a bad state transition or something, the lock would never fully be released because the following unlock isn't in a finally.